src/Instantiator/ObjectInstantiator.php

```diff
@@ -36,30 +36,28 @@ public function __construct($factory)
      */
     public function instantiateObject()
     {
-        if ($this->factory instanceof \Closure) {
-            return ($this->factory)($this->getData());
-        }
-
-        if (\is_string($this->factory)) {
+        if (\is_string($this->factory) && class_exists($this->factory)) {
```

Comment:
Do we really need this additional check?

Reply:
yes, since we don't know the type of `$this->factory` it can be `class-string|\Closure|(callable&array)` AND `class_exists` expects string.

It can be done the way I did OR the way below (move `class_exists` check to the end)

```php
        if (\is_array($this->factory) && \is_callable($this->factory)) {
            $class = \is_object($this->factory[0]) ? \get_class($this->factory[0]) : $this->factory[0];
            $factoryMethod = (new \ReflectionMethod($class, $this->factory[1]));
            $factoryMethodAsString = $class.'::'.$this->factory[1];
            if (!$factoryMethod->isPublic()) {
                throw new TransformationFailedException(sprintf('The factory method %s() is not public.', $factoryMethodAsString));
            }
        } elseif ($this->factory instanceof \Closure) {
            $factoryMethod = new \ReflectionFunction($this->factory);
        } elseif (class_exists($this->factory)) {
            $factoryMethod = (new \ReflectionClass($this->factory))->getConstructor();
            $factoryMethodAsString = $this->factory.'::__construct';
            if (null === $factoryMethod) {
                throw new TransformationFailedException(sprintf('The class "%s" used as a factory does not have a constructor.', $this->factory));
            }
            if (!$factoryMethod->isPublic()) {
                throw new TransformationFailedException(sprintf('The factory method %s() is not public.', $factoryMethodAsString));
            }
        } else {
            return $this->getData();
        }
```

BUT since you're saying 
> keep the diff as small as possible
I will leave it the way I did

Reply:
I was mainly wondering if we were missing that before (no need to revert in that case by the way I am just trying to understand :)) or if it was needed because of the other changes here.

Reply:
PHPStan is yelling at me if I do

```
if (class_exists($this->factory)) {
```

Reply:
Well, but why do we need the `class_exists()` call at all? Does anything fail without it?

Reply:
Nothing fails.

Thanks for suggestion